### PR TITLE
BZ-18892: feat: add an option to include icon within select component

### PR DESCRIPTION
**Pull request by adarshjuspay**
- add an option to include icon within the select component
 - implement CSS styles to ensure proper alignment and display of icon

### CODE REVIEW

File: src/lib/Select/Select.svelte [hunk 1]
```diff
@@ -139,6 +141,11 @@
       role="button"
       tabindex="0"
     >
+    {#if properties.icon && properties.iconPath}
```
Comment (murdore) [inline src/lib/Select/Select.svelte]: add empty string check or default to null above

File: src/lib/Select/Select.svelte [hunk 2]
```diff
@@ -139,6 +141,11 @@
...
+      <div class="icon-container">
+        <img src={properties.iconPath} class="icon" alt="" />
```
Comment (murdore) [inline src/lib/Select/Select.svelte]: expose alt via input

File: src/lib/Select/Select.svelte [hunk 3]
```diff
@@ -216,8 +223,8 @@
   }
 
   .arrow {
-    height: 16px;
-    width: 16px;
+    height:var(--arrow-height, 16px);
```
Comment (murdore) [inline src/lib/Select/Select.svelte]: rename to dropdown-icon-height and width

File: src/lib/Select/Select.svelte [hunk 4]
```diff
@@ -13,7 +13,9 @@
     selectedItem: '',
     selectedItemLabel: null,
     showSelectedItemInDropdown: false,
-    selectMultipleItems: false
+    selectMultipleItems: false,
+    icon:false,
+    iconPath:''
```
Comment (murdore) [inline src/lib/Select/Select.svelte:18]: do we need to props?

**Review by murdore — CHANGES_REQUESTED** [review]

**Comment by murdore** [comment]
@adarshjuspay design is not matching with attached screenshot

File: src/lib/Select/Select.svelte (hunk 4 above)
Comment (adarshjuspay) [reply src/lib/Select/Select.svelte:18]: resolved

File: src/lib/Select/Select.svelte (hunk 1 above)
Comment (adarshjuspay) [reply src/lib/Select/Select.svelte]: done

File: src/lib/Select/Select.svelte (hunk 2 above)
Comment (adarshjuspay) [reply src/lib/Select/Select.svelte]: done

File: src/lib/Select/Select.svelte (hunk 3 above)
Comment (adarshjuspay) [reply src/lib/Select/Select.svelte]: done

**Comment by adarshjuspay** [comment]
> @adarshjuspay design is not matching with attached screenshot

dev proof updated

**Review by ARX1718 — APPROVED** [review]

**Comment by rab1prasad** [comment]
This pr has 2 commits, can we do this here?

File: src/lib/Select/Select.svelte [hunk 5]
```diff
@@ -6,14 +6,16 @@
 
   let selectedElementDiv: HTMLDivElement | null = null;
 
+  export let alt='';
```
Comment (rab1prasad) [inline src/lib/Select/Select.svelte:9]: Can we have this `alt` text in `SelectProperties`?

File: src/lib/Select/properties.ts [hunk 6]
```diff
@@ -6,4 +6,5 @@ export type SelectProperties = {
   selectedItemLabel: string | string[] | null;
   showSelectedItemInDropdown: boolean;
   selectMultipleItems: boolean;
+  iconPath:string;
```
Comment (rab1prasad) [inline src/lib/Select/properties.ts]:
```suggestion
  iconPath:string | null;
```

File: src/lib/Select/Select.svelte [hunk 7]
```diff
@@ -139,6 +141,11 @@
       role="button"
       tabindex="0"
     >
+    {#if properties.iconPath.length!==0}
```
Comment (rab1prasad) [inline src/lib/Select/Select.svelte]:
```suggestion
    {#if properties.iconPath !== null && properties.iconPath !== ""}
```

File: src/lib/Select/Select.svelte [hunk 8]
```diff
@@ -141,9 +141,9 @@
       role="button"
       tabindex="0"
     >
-    {#if properties.icon && properties.iconPath}
+    {#if properties.iconPath.length!==0}
```
Comment (murdore) [inline src/lib/Select/Select.svelte]: add null check

File: src/lib/Select/Select.svelte (hunk 5 above)
Comment (adarshjuspay) [reply src/lib/Select/Select.svelte:9]: we can take alt directly as component prop

**Review by rab1prasad — APPROVED** [review]

**Review by juspay-aayush — APPROVED** [review]

**Review by vishalbhardwaj7 — APPROVED** [review]

**Review by ikra2407 — APPROVED** [review]

**Review by nirupam-juspay — APPROVED** [review]

**Review by kingk5 — APPROVED** [review]

**Review by Ishleenkaurkapoor — APPROVED** [review]

**Review by murdore — APPROVED** [review]

**Pull request merged**